src/printer.js

```diff
@@ -3880,7 +3880,28 @@ function makeString(rawContent, enclosingQuote) {
     return match;
   });
 
-  return enclosingQuote + newContent + enclosingQuote;
+  // Matches any unnecessarily escaped character.
+  // Adapted from https://github.com/eslint/eslint/blob/de0b4ad7bd820ade41b1f606008bea68683dc11a/lib/rules/no-useless-escape.js#L27
+  const regexUnnecessaryStringEscapes = /((?:^|[^\\])(?:\\\\)*)\\([^\\nrvtbfux\r\n\u2028\u2029"'0-7])/g;
+
+  let minimallyEscapedContent = newContent;
+  let minimallyEscapedContentPrev;
```

Comment:
Heh, oops. Times like these make me wish this project had automated linting :P 

Fixed in https://github.com/prettier/prettier/pull/1575/commits/8cb1e1a1437f568466929c99fc080dc21eb2ad04